pkg/storage/module/upload.go

```diff
@@ -14,16 +15,42 @@ import (
 const numFiles = 3
 
 // Uploader takes a stream and saves it to the blob store under a given path
-type Uploader func(ctx context.Context, path, contentType string, stream io.Reader) error
+type Uploader func(ctx context.Context, path, contentType string, stream Stream) error
+
+// Stream is the object that is passed along to Uploaders
+type Stream struct {
+	Stream io.Reader
+	Size   int64
+}
+
+// NewStreamFromBytes returns a new module.Stream from a slice of bytes to Savers
+func NewStreamFromBytes(b []byte) Stream {
+	return Stream{
+		Stream: bytes.NewReader(b),
```

Comment:
why dont you have a check for size over limit resulting in `-1` in here as well?